### ISSUE TITLE
Request: Default values for template placeholders

### ISSUE THREAD

**Issue by archetyped** [websharks/wp-snippets#5]
Hi, it would be great if we could set default values for placeholders (e.g. `%%type%%`) when creating a snippet template.  This would allow the snippet output to still make sense even if one of placeholders was not defined when using the snippet template.

**Comment by raamdev** [comment]
@archetyped Could you explain what you mean by a "snippet template"? If you could explain the scenario where you'd use what you're referring to, it might help me get a better idea what you're asking. :-)

**Comment by archetyped** [comment]
Standard Snippet: `[snippet slug="test"]`
Snippet template: `[snippet_template slug="test-template"]`

**Comment by raamdev** [comment]
@archetyped I see. So the Snippet Template would essentially allow you to build a snippet that is a collection of other snippets?

`[snippet_template slug="test-template"]` contents:

```
This is a template that includes another snippet, like %%test%%.
```

That snippet template would then include the contents of `[snippet slug="test"]` where the `%%test%%` replacement tag appears, correct?

**Comment by archetyped** [comment]
No.  Here's the [existing code for processing snippet templates](https://github.com/websharks/wp-snippets/blob/000000-dev/wp-snippets/wp-snippets.php#L267-L277).  Once you have a look at that, my request should make more sense.

**Comment by raamdev** [comment]
@archetyped I'm familiar with the code, but I'm still not understanding what you're asking.

Can you give me an example of how you would set a default value in the shortcode?

**Comment by raamdev** [comment]
@archetyped Are you thinking something like this? `%%type|default value goes here%%`

So that when the template is processed and WP Snippets is done replacing all the attributes with their values, WP Snippets can go through again looking for any remaining replacement tags (which did not contain any assigned values in the shortcode) and replace those with their default values? 

So in this case, if we called `[snippet_template slug="test" country="USA"]`, the `%%country%%` would get replaced with `USA` but since the shortcode did not contain any `type=""` assignment the `%%type%%` tag would simply get replaced with `default value goes here`, since that's its default value.

Do I have that right?

**Comment by archetyped** [comment]
Yes, that is correct.

**Comment by raamdev** [comment]
Cool. Thanks for the feature request! I'll mark this as an enhancement and add it to the todo list for a future release. :-)
